README.md

```diff
@@ -82,6 +82,85 @@ provided as repository secrets.
           password: ${{ secrets.QUAY_PASSWORD }}
 ```
 
+### Java setup
+
+#### Setup JDK
+
+[actions/setup-java](https://github.com/actions/setup-java) should be used, here is a sample usage:
+
+```yml
+      - name: Set up JDK 11
+        uses: actions/setup-java@v3
+        with:
+          java-version: '11'
+          distribution: 'temurin'
+          cache: 'maven'
+```
+
+#### Setup Maven Credentials
+
+Credentials should be already available via organization secrets, otherwise they would need to be
+provided as repository secrets.
+
+Since repositories hold a `settings.xml` file at the root with environment variables `MAVEN_USERNAME` and
+`MAVEN_USERNAME` filled for the username and password, only a mapping of variables is needed:
+
+```yml
+      - name: Build with Maven
+        run: mvn --settings settings.xml [...]
+        env:
+          MAVEN_USERNAME: ${{ secrets.NEXUS_USERNAME }}
+          MAVEN_USERNAME: ${{ secrets.NEXUS_PASSWORD }}
+```
+
+Alternatively, the [s4u/maven-settings-action](https://github.com/s4u/maven-settings-action) could be used.
+
+#### Setup Maven Build Options
+
+Maven build options can be shared for a given step on the mvn command line, or extracted as environment variables.
+
+Sample usage:
+
+```yml
+      - name: Test with Maven
+        run: mvn verify ${{ env.MAVEN_CLI_OPTS}}
+        env:
+          MAVEN_CLI_OPTS: --show-version -Ddocker.skip -Dlogging.root.level=off -Dspring.main.banner-mode=off
+```
+
+When deploying in a second step, these variables can be shared:
+
+```yml
+env:
+  MAVEN_CLI_OPTS: --show-version -Dlogging.root.level=off -Dspring.main.banner-mode=off
+
+[...]
+
+      - name: Test with Maven
+        run: mvn verify ${{ env.MAVEN_CLI_OPTS}}
+      - name: Deploy with Maven
+        run: mvn deploy ${{ env.MAVEN_CLI_OPTS}} -DskipTests
+```
+
+When migrating from Travis, depending on the previous configuration, docker.skip and docker.tag properties might need
+to be setup on the command line.
+
+Here is a sample way to extract a branch name that would be used for docker images built with the `build-and-push-docker-images.sh` script, although using the [dedicated action](#dockerbuild-push-action) can also be
+useful.
+
+```yml
+      - name: Get the current branch name
+        id: branch
+        shell: bash
+        run: echo "::set-output name=name::${GITHUB_REF#refs/heads/}"
+      - name: Set stripped branch name as tag
+        run: echo "TAG=$(echo ${{ steps.branch.outputs.name }} | sed -e 's/[^-_.[:alnum:]]/_/g')" >> $GITHUB_ENV
```

Comment:
latest GHA updates introduced $GITHUB_REF_NAME so it's necessary anymore to extract branch name with that

Reply:
Please check my update (not sure of the mapping of these in env)